### PR TITLE
Feat: CloudFunction to ElasticSearch Integration

### PR DESCRIPTION
**Pull request by dikshamodi**
This cloud function integrates pubsub to Elasticsearch. Data pushed to pubsub stream by the Spring App triggers the cloud function. Once triggered, the cloud function fetches the latest shop, merchant and catalog details and packages it into a JSON format according to the mapping specified for elasticsearch index (mapping is added in this PR too)

The data is then posted to elasticsearch index where it is inserted/updated. The ID is kept the same as shopID

### CODE REVIEW

File: cloudfunction/pushToElasticSearch.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+* @param {!Object} event Event payload.
+* @param {!Object} context Metadata for the event.
+*/
+
+const mysql = require('promise-mysql');
+const axios = require('axios');
+const SHOP_SELECT_STATEMENT = `SELECT MerchantID, ShopID, ShopName, Latitude, Longitude, AddressLine1, TypeOfService FROM Shops WHERE ShopID=?;`;
+const MERCHANT_SELECT_STATEMENT = `SELECT MerchantName, MerchantPhone FROM Merchants WHERE MerchantID=?;`
+const CATALOG_SELECT_STATEMENT = `SELECT ServiceName from Catalog WHERE ShopID = ?;`
+const GCE_URL = 'http://10.128.0.13:9200'
```
Comment (harshil93) [inline cloudfunction/pushToElasticSearch.js:13]: Can you comment or make the variable name more elaborate for this. GCE_URL is pretty generic and doesn't tell anything

File: cloudfunction/pushToElasticSearch.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+exports.elasticsearch = async (event, context) => {
+  const shopID = Buffer.from(event.data, 'base64').toString()
+  connection = await mysql.createConnection(dbConfig);
+  const shopData = await getShopDataByShopID(shopID);
+  const merchantData = await getMerchantDataByMerchantID(shopData[0].MerchantID);
+  const catalogItems = await getCatalogByShopID(shopID);
+  shopDataJson = createElasticSearchJson(shopData[0], merchantData[0], catalogItems);
+  pushToElasticsearch(shopDataJson);
```
Comment (harshil93) [inline cloudfunction/pushToElasticSearch.js:91]: Won't you have to await on this function ?

File: cloudfunction/pushToElasticSearch.js (hunk 1 above)
Comment (dikshamodi) [reply cloudfunction/pushToElasticSearch.js:13]: Okay. Will do that. Will ELASTICSEARCH_SERVER_URL be better?

File: cloudfunction/pushToElasticSearch.js (hunk 2 above)
Comment (dikshamodi) [reply cloudfunction/pushToElasticSearch.js:91]: We can await. I skipped await since that was the last statement. Will it be good to add it?

**Pull request merged**